lib/ansible/modules/files/blockinfile.py

```diff
@@ -282,6 +282,8 @@ def main():
             n0 = i
         if line == marker1:
             n1 = i
+            if (n0 != None):
+              break
```

Comment:
I would not change current behaviour, add an option that defaults to the current `greedy=yes|no`

Reply:
> I would not change current behaviour, add an option that defaults to the current `greedy=yes|no`

@bcoca
That would be wonderful to have such an option. Thank you for your adoption.